ingest/importer/importer.py

```diff
@@ -42,37 +40,36 @@ def _generate_spreadsheet_json(self, file_path, project_uuid=None):
                 f'There was an error retrieving the schema information to process the spreadsheet. {str(e)}')
 
         workbook_importer = WorkbookImporter(template_mgr)
-        spreadsheet_json = workbook_importer.do_import(ingest_workbook, project_uuid)
+        spreadsheet_json, errors = workbook_importer.do_import(ingest_workbook, project_uuid)
 
-        return spreadsheet_json, template_mgr
+        return spreadsheet_json, template_mgr, errors
 
-    # TODO nothing seems to be using the project_uuid argument. Why is this even here?
     def import_file(self, file_path, submission_url, project_uuid=None):
-        error_json = None
         submission = None
         try:
-            spreadsheet_json, template_mgr = self._generate_spreadsheet_json(file_path, project_uuid)
-            entity_map = self._process_links_from_spreadsheet(template_mgr, spreadsheet_json)
-
-            submitter = IngestSubmitter(self.ingest_api)
-
-            # TODO the submission_url should be passed to the IngestSubmitter instead
-            submission = submitter.submit(entity_map, submission_url)
-
-        except ingest.importer.submission.Error as e:
-            error_json = SubmissionError(str(e)).getJSON()
-            self.logger.error(str(e), exc_info=True)
+            spreadsheet_json, template_mgr, errors = self._generate_spreadsheet_json(file_path, project_uuid)
+            if not errors:
+                entity_map = self._process_links_from_spreadsheet(template_mgr, spreadsheet_json)
+
+                # TODO the submission_url should be passed to the IngestSubmitter instead
+                submitter = IngestSubmitter(self.ingest_api)
+                submission = submitter.submit(entity_map, submission_url)
+                self.logger.info(f'Submission in {submission_url} is done!')
+            else:
+                self.report_errors(submission_url, errors)
         except Exception as e:
-            error_json = ImporterError(str(e)).getJSON()
+            self.ingest_api.create_submission_error(submission_url, ImporterError(str(e)).getJSON())
             self.logger.error(str(e), exc_info=True)
-
-        if error_json:
-            self.ingest_api.create_submission_error(submission_url, error_json)
-        else:
-            self.logger.info(f'Submission in {submission_url} is done!')
-
         return submission
 
+    def report_errors(self, submission_url, errors):
+        for error in errors:
+            self.ingest_api.create_submission_error(
+                submission_url,
+                ParserError(error["location"], error["type"], error["detail"]).getJSON()
+            )
+            self.logger.error(error["detail"], exc_info=True)
```

Comment:
I think this line has the potential to make our logs really big, especially since we're now aggregating errors

Reply:
Since we've already handled these "errors", maybe we don't need to log them. Arguably they're not errors in the system